packages/smooth_app/lib/temp/user_preferences.dart

```diff
@@ -41,66 +41,54 @@ class UserPreferences extends ChangeNotifier {
     PantryType.SHOPPING: _TAG_SHOPPING_REPOSITORY,
   };
 
-  Future<void> init(final UserPreferencesModel userPreferencesModel) async {
+  Future<void> init(final ProductPreferences productPreferences) async {
     final bool alreadyDone = _sharedPreferences.getBool(_TAG_INIT);
     if (alreadyDone != null) {
       return;
     }
-    await resetImportances(userPreferencesModel);
+    await resetImportances(productPreferences);
     await _sharedPreferences.setBool(_TAG_INIT, true);
   }
 
   String _getImportanceTag(final String variable) =>
       _TAG_PREFIX_IMPORTANCE + variable;
 
-  Future<void> setImportance(String variable, int value) async =>
-      _setImportance(variable, value, notify: true);
-
-  Future<void> _setImportance(
-    String variable,
-    int value, {
-    bool notify = true,
-  }) async {
-    await _sharedPreferences.setInt(_getImportanceTag(variable), value);
-    if (notify) {
-      notifyListeners();
-    }
-  }
+  Future<void> setImportanceIndex(
+          final String attributeId, final int importanceIndex) async =>
+      await _sharedPreferences.setInt(
+          _getImportanceTag(attributeId), importanceIndex);
 
-  int getImportance(String variable) =>
-      _sharedPreferences.getInt(_getImportanceTag(variable)) ??
+  int getImportanceIndex(final String attributeId) =>
+      _sharedPreferences.getInt(_getImportanceTag(attributeId)) ??
       PreferenceImportance.INDEX_NOT_IMPORTANT;
 
   Future<void> resetImportances(
-      final UserPreferencesModel userPreferencesModel) async {
+    final ProductPreferences productPreferences,
+  ) async {
     for (final AttributeGroup attributeGroup
-        in userPreferencesModel.attributeGroups) {
+        in productPreferences.attributeGroups) {
       for (final Attribute attribute in attributeGroup.attributes) {
-        await _setImportance(
-          attribute.id,
-          PreferenceImportance.INDEX_NOT_IMPORTANT,
-          notify: false,
-        );
+        await setImportanceIndex(
+            attribute.id, PreferenceImportance.INDEX_NOT_IMPORTANT);
       }
     }
-    int valueIndex;
-    valueIndex = userPreferencesModel
-        .getValueIndex(PreferenceImportance.ID_VERY_IMPORTANT);
-    if (valueIndex != null) {
-      await _setImportance(
-          UserPreferencesModel.ATTRIBUTE_NUTRISCORE, valueIndex,
-          notify: false);
+    int importanceIndex;
```

Comment:
We probably should separate clearPreferences (-> mark everything as unimportant) and resetPreferences (put back default preferences). clearPreferences should be in openfoodfacts-dart, but resetPreferences should be app specific. A vegan app for instance would have different default settings.

Reply:
`UserPreferences` (where `resetPreferences` is) is _not_ supposed to go to OFF-dart - though it's currently in a `temp` folder.
I need to add `clearPreferences` (in `ProductPreferences`).